ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java

```diff
@@ -51,23 +76,46 @@ protected void doHandleException(final Throwable t) {
     }
   }
 
-  protected abstract void handleMetadata(QueryResponseMetadata queryResponseMetadata);
+  protected abstract void handleMetadata(StreamedRow queryResponseMetadata);
 
-  protected abstract void handleRow(Buffer buff);
+  protected abstract void handleRow(StreamedRow buff);
 
   protected abstract void handleExceptionAfterFutureCompleted(Throwable t);
 
   private void handleArgs(final Buffer buff) {
     hasReadArguments = true;
 
-    final QueryResponseMetadata queryResponseMetadata;
+    final Buffer strippedValidJson = toValidJsonMsg(buff, true);
+    final StreamedRow queryResponseMetadata;
     try {
-      queryResponseMetadata = JSON_MAPPER.readValue(buff.getBytes(), QueryResponseMetadata.class);
+      queryResponseMetadata = JSON_MAPPER.readValue(
+              strippedValidJson.getBytes(),
+              StreamedRow.class);
+
     } catch (Exception e) {
       cf.completeExceptionally(e);
       return;
     }
 
     handleMetadata(queryResponseMetadata);
   }
+
+  public static Buffer toValidJsonMsg(final Buffer responseLine, final boolean stripArray) {
```

Comment:
Not a big deal, but this is copied logic.  Might makes sense to put in common if you think we'll modify this much.

Reply:
done